[DOCS-1690] Add docs for Restricted Viewer role (#1565)
wandb/docs: content/en/guides/core/registry/aliases.md

- You might use aliases such as `dataset_version_v0`, `dataset_version_v1`, and `dataset_version_v2` to identify which dataset a model was trained on.
- You might use a `best_model` alias to keep track of the best performing artifact model version.

> Any user with a [**Member** or **Admin** registry role]({{< relref "guides/core/registry/configure_registry/#registry-roles" >}}) on a registry can add or remove a custom alias from a linked artifact in that registry. Users with the **Restricted Viewer** or **Viewer** roles cannot add or remove aliases.

Reviewer: Suggestion: Also link Restricted and View to "guides/core/registry/configure_registry/#registry-roles"

> Any user with a [**Member** or **Admin** registry role]({{< relref "guides/core/registry/configure_registry/#registry-roles" >}}) on a registry can add or remove a custom alias from a linked artifact in that registry. Users with the **Restricted Viewer** or **Viewer** roles cannot add or remove aliases.

Author: ```suggestion
Any user with a [**Member** or **Admin** registry role]({{< relref "guides/core/registry/configure_registry/#registry-roles" >}}) on a registry can add or remove a custom alias from a linked artifact in that registry. Users with the [**Restricted Viewer** or **Viewer** roles]({{< relref "guides/core/registry/configure_registry/#registry-roles" >}}) cannot add or remove aliases.
```

- You might use aliases such as `dataset_version_v0`, `dataset_version_v1`, and `dataset_version_v2` to identify which dataset a model was trained on.
- You might use a `best_model` alias to keep track of the best performing artifact model version.

Any user with a [**Member** or **Admin** registry role]({{< relref "guides/core/registry/configure_registry/#registry-roles" >}}) on a registry can add or remove a custom alias from a linked artifact in that registry. Users with the **Restricted Viewer** or **Viewer** roles cannot add or remove aliases.
[Protected aliases]({{< relref "aliases/#protected-aliases" >}}) provide a way to label and identify which artifact versions to protect from modification or deletion.

{{% alert %}}

> [Protected aliases]({{< relref "aliases/#protected-aliases" >}}) provide a way to label and identify which artifact versions to protect from modification or deletion.

Reviewer 2: **question (non-blocking):** Was the snippet about protected aliases intentionally added twice here?

Author: No, I was dithering about having it in an alert or not and didn't clean it up. Thanks.

- You might use aliases such as `dataset_version_v0`, `dataset_version_v1`, and `dataset_version_v2` to identify which dataset a model was trained on.
- You might use a `best_model` alias to keep track of the best performing artifact model version.

> Any user with a [Member or Admin registry role]({{< relref "guides/core/registry/configure_registry/#registry-roles" >}}) on a registry can add or remove a custom alias from a linked artifact in that registry. **Restricted Viewer** users cannot add or remove aliases. If appropriate, use [protected aliases]({{< relref "aliases/#protected-aliases" >}}) to label and identify which artifact versions to protect from modification or deletion.

Author: ```suggestion
Any user with a [Member or Admin registry role]({{< relref "guides/core/registry/configure_registry/#registry-roles" >}}) on a registry can add or remove a custom alias from a linked artifact in that registry. Users with the **Restricted Viewer** or **Viewer** roles cannot add or remove aliases.

[Protected aliases]({{< relref "aliases/#protected-aliases" >}}) provide a way to label and identify which artifact versions to protect from modification or deletion.
```
Convert the last sentence to a note

### Protected aliases
Use a [protected alias]({{< relref "aliases/#protected-aliases" >}}) to both label and identify artifact versions that should not be modified or deleted. For example, consider using a `production` protected alias to label and identify artifact versions that are in used in your organization's machine learning production pipeline.

> [Registry admins]({{< relref "/guides/core/registry/configure_registry/#registry-roles" >}}) and [service accounts]({{< relref "/support/kb-articles/service_account_useful" >}}) with the **Admin** role can create protected aliases and add or remove protected aliases from an artifact version. **Members** and **Viewers** cannot unlink a protected version or delete a collection that contains a protected . See [Configure registry access]({{< relref "/guides/core/registry/configure_registry.md" >}}) for details.

Reviewer 2: **suggestion (non-blocking):** Should the `Restricted viewer` role be mentioned here too?

Author: Done

1. First, you create an artifact object (`wandb.Artifact()`).
2. Next, you add two dataset PyTorch tensors to the artifact object with `wandb.Artifact.add_file()`. 

> 3. Lastly, you link the artifact version to the `Model` collection in the `Zoo_Classifier_Models` registry with `link_artifact()`. You also add two custom aliases to the artifact version by passing  `production-us` and `production-eu` as arguments to the `aliases` parameter.

Reviewer: Why this highlighted in the PR review?

Author: Whitespace, A newline character was added at the end of the file. For some reason turning off whitespace in the diff viewer doesn't change how this line shows up. I asked Cursor locally because Copilot could not help me.
